Nautilus/Handlers/ModDataBankHandler.cs

```diff
@@ -0,0 +1,114 @@
+using BepInEx;
+using Nautilus.Utility;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Nautilus.Handlers;
+public class ModDataBankHandler
+{
+    private static bool isinit;
+    private static List<PDAEncyclopedia.EntryData> waitlist = new();
+    internal static void Initialize(uGUI_EncyclopediaTab tab)
+    {
+        LanguageHandler.SetLanguageLine("EncyPath_Mods", "Mods");
+        isinit = true;
+        foreach(var data in waitlist)
+        {
+            CompleteRegister(data);
+        }
+    }
+
+    private static void CompleteRegister(PDAEncyclopedia.EntryData data)
+    {
+        if (isinit)
+        {
+            InternalLogger.Info($"{data.key} entry added.");
+            PDAHandler.AddEncyclopediaEntry(data);
+            PDAEncyclopedia.Add(data.key, true);
+        }else
+        {
+            waitlist.Add(data);
+        }
+
+    }
+    /// <summary>
+    /// Register mod with default values. 
+    /// </summary>
+    /// <param name="assembly">Assembly for your mod.</param>
+    public static void RegisterMod(Assembly assembly)
+    {
+        var bepinplugin = assembly.GetTypes().First(type => type.GetCustomAttributes(false).OfType<BepInPlugin>().Any());
+        var bepinplugindata = bepinplugin.GetCustomAttributes(false).OfType<BepInPlugin>().FirstOrDefault();
```

Comment:
Instead of using reflection off the assembly it might make more sense to just use the chainloader and have mods pass their GUID?

https://docs.bepinex.dev/v5.4.21/api/BepInEx.Bootstrap.Chainloader.html#BepInEx_Bootstrap_Chainloader_PluginInfos

Reply:
I feel that there wouldn't be much of a performance difference, but I'll do it anyways.

Reply:
In terms of performance, I imagine it would be a bit faster as it's a single dictionary lookup rather than an iteration of the types. But vastly more importantly, it's simpler IMO.

Reply:
Should be fixed.